controllers/profile.js

```diff
@@ -0,0 +1,48 @@
+const mongoose = require('mongoose');
+const { logger } = require('../logging/logger');
+
+const Users = mongoose.model('User');
+
+// WARNING any schema properties in this object will be exposed to other clients,
+// do not put sensitive schema properties in this object
+const PUBLIC_CONTACT_SCHEMA = {
+    user_type: 1,
+    person: 1,
+    demo: 1,
+    gen_interest: 1,
+    spec_interests: 1,
+    sports: 1,
+};
+
+exports.getProfileData = function (req, res) {
+    return res.status(200).json(req.user);
+};
+
+exports.getContacts = function (req, res) {
+    if (req.user.user_type === 'mentor') {

```

Comment:
I ended up experimenting with this, and this is the approach I ended up with. What are your thoughts on this approach? I like that it is agnostic to the user type. 


```exports.getContacts = function (req, res) {
    logger.info("Getting contacts");
    Users.findById(req.user._id, { contacts: 1 })
        .populate('contacts', PUBLIC_CONTACT_SCHEMA)
        .then((result) => {
            result.contacts = result.contacts.map((contact) => {
                if (contact.contacts.indexOf(req.user._id) < 0) {
                    logger.info("This user should not be returned");
                    return null;
                }
                return contact;
            });
            return result;
        })
        .then(contacts => res.json(contacts))
        .catch((err) => {
            logger.info("An error occurred while processing a getContacts request");
            logger.error(err);
            logger.error(err.stack);
            res.status(500).json({ Error: 'Something went wrong' });
        });
};

Reply:
I'm not exactly sure what you're trying to do here. Are you determining whether a users contacts have that user in their contacts? We kind of have to assume that's the case, if that weren't the case we would have a far larger data integrity problem. Getting a users contacts and  creating a contact link between users should be independent operations

Reply:
Also this doesn't really address the issue of the mentor-mentee relationship exposing different data than the other contact relationships

Reply:
Yeah, I realize now that I had an early misunderstanding, and then I haven't been precise enough in my explanation.  

My original concern was that I want to make sure that the only users who can see user A's profile information are users who are "connected" to user A. This doesn't really matter whether they are a mentor-mentee or mentee-mentee relationship, etc. 

In the original API design, you would call getProfile and pass a userId, so we needed to prevent a user who doesn't have a relationship with user A from calling getProfile/userA. The check we were doing back then was basically what I have here, which is, "is the calling user in the contacts of the located user".

In your new version, you can only ever call getContacts on the calling user, which is validated against the api-key passed. I'm only just now realizing that means we will never have the problem I describe above of an unknown user calling for the profile information on user A. 

All of this is a roundabout way to say that your original code, before I made you add any of the public schema stuff, etc. is correct, and prevents us from leaking profile information to users who are not "connected" to a user.

Reply:
Yeah although we still need the `PUBLIC_CONTACT_SCHEMA` as it defines what field the mongoose populate command populates, as we obviously don't want to expose certain fields to another client. I have tests for both the mentor and mentee cases in `endpoint_test.js` if you want an example of what I'm talking about.

Reply:
Yeah, that makes sense. The only thing we need to add to public schema is "school", since that also gets displayed in the client app

Reply:
This change has already been made

Reply:
Okay cool. Does it make sense to go back to the populate() you had before where we just straight populate the public schema and return?